js/jquery.emojiarea.js

```diff
@@ -827,14 +837,34 @@
           }
         }
 
+        /**
+         * @private
+         * @function onOpenEmojiMenuElement
+         *
+         * @description Called when the emoji menu is about to open.  This function is used to reset the
+         * insertEmojiFunc based on the scope of the element.  For this to work the targetIdentifier needs
+         * to be set on the text area.
+         *
+         */
+        function onOpenEmojiMenuElement(evt, targetIdentifier) {
+          if(targetIdentifier !== attrs["targetIdentifier"]){
+            return;
+          }
+
+          // update the emoji insert function with this scope's function
+          window.emojiPicker.insertEmojiFunc = scope.insertEmojiFunc;
+        }
+
         // I am not sure if the name open.leave-conf-dialog, is generic enough, but
         // I guess it is OK.
         scope.$on('open.leave-conf-dialog', hideEmojiMenuElement)
+        scope.$on('bu.emoji-menu.open', onOpenEmojiMenuElement)
```

Comment:
I suggest changing the event name to emoji-menu.open to keep it independent of bublup. Note that this is a public repository.